src/sentry/api/endpoints/group_events.py

```diff
@@ -38,63 +43,78 @@ def get(self, request, group):
         :pparam string issue_id: the ID of the issue to retrieve.
         :auth: required
         """
+        use_snuba = options.get('snuba.events-queries.enabled')
+        backend = self._get_events_snuba if use_snuba else self._get_events_legacy
+        return backend(request, group)
 
-        def respond(queryset):
-            return self.paginate(
-                request=request,
-                queryset=queryset,
-                order_by='-datetime',
-                on_results=lambda x: serialize(x, request.user),
-                paginator_cls=DateTimePaginator,
-            )
-
-        events = Event.objects.filter(group_id=group.id)
+    def _get_events_snuba(self, request, group):
+        from functools32 import partial
+        from sentry.api.paginator import GenericOffsetPaginator
+        from sentry.api.serializers.models.event import SnubaEvent
+        from sentry.utils.snuba import raw_query
 
         try:
-            environment = self._get_environment_from_request(
-                request,
-                group.project.organization_id,
-            )
-        except Environment.DoesNotExist:
-            return respond(events.none())
-
-        raw_query = request.GET.get('query')
-
-        if raw_query:
-            try:
-                query_kwargs = parse_query([group.project], raw_query, request.user)
-            except InvalidQuery as exc:
-                return Response({'detail': six.text_type(exc)}, status=400)
+            environment = self._get_environment(request, group)
```

Comment:
Does it make sense to move these calls outside and to pass the variables in?

Reply:
you mean move them up into the `get()`? yeah that would probably make sense and be a little DRYer

Reply:
done